Index metric type SUBSTRUCTURE does not match index type BIN_FLAT

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```
___ TestLoadBase.test_load_collection_after_index_binary[get_binary_index0] ____

self = <test_load_collection.TestLoadBase object at 0x7f263c7f3da0>
connect = <milvus.client.stub.Milvus object at 0x7f2655f8b470>
binary_collection = 'load_collection_TRDnQx0B'
get_binary_index = {'index_type': 'BIN_FLAT', 'metric_type': 'SUBSTRUCTURE', 'params': {'nlist': 128}}

    @pytest.mark.level(2)
    def test_load_collection_after_index_binary(self, connect, binary_collection, get_binary_index):
        '''
        target: test load binary_collection, after index created
        method: insert and create index, load binary_collection with correct params
        expected: no error raised
        '''
        connect.insert(binary_collection, binary_entities)
        connect.flush([binary_collection])
        for metric_type in binary_metrics():
            logging.getLogger().info(metric_type)
            get_binary_index["metric_type"] = metric_type
            if get_binary_index["index_type"] == "BIN_IVF_FLAT" and metric_type in structure_metrics():
                with pytest.raises(Exception) as e:
                    connect.create_index(binary_collection, binary_field_name, get_binary_index)
            else:
>               connect.create_index(binary_collection, binary_field_name, get_binary_index)

collection/test_load_collection.py:74: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.6/site-packages/milvus/client/stub.py:35: in inner
    return func(self, *args, **kwargs)
/usr/local/lib/python3.6/site-packages/milvus/client/stub.py:452: in create_index
    return handler.create_index(collection_name, field_name, params, timeout, **kwargs)
/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py:50: in handler
    raise e
/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py:42: in handler
    return func(self, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <milvus.client.grpc_handler.GrpcHandler object at 0x7f2655f8b3c8>
collection_name = 'load_collection_TRDnQx0B', field_name = 'binary_vector'
params = {'index_type': 'BIN_FLAT', 'metric_type': 'SUBSTRUCTURE', 'params': {'nlist': 128}}
timeout = None, kwargs = {}
index_param = collection_name: "load_collection_TRDnQx0B"
field_name: "binary_vector"
extra_params {
  key: "index_type"
  value: "B...ra_params {
  key: "params"
  value: "{\"nlist\":128}"
}
extra_params {
  key: "metric_type"
  value: "SUBSTRUCTURE"
}

future = <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.OK
	details = ""
>
status = error_code: ILLEGAL_ARGUMENT
reason: "Index metric type SUBSTRUCTURE does not match index type BIN_FLAT"


    @error_handler()
    def create_index(self, collection_name, field_name, params, timeout=None, **kwargs):
        index_param = Prepare.index_param(collection_name, field_name, params)
        future = self._stub.CreateIndex.future(index_param, wait_for_ready=True, timeout=timeout)
        if kwargs.get('_async', False):
            cb = kwargs.get("_callback", None)
            return CreateIndexFuture(future, cb)
        status = future.result()
    
        if status.error_code != 0:
>           raise BaseException(status.error_code, status.reason)
E           milvus.client.exceptions.BaseException: <BaseException: (code=5, message=Index metric type SUBSTRUCTURE does not match index type BIN_FLAT)>

/usr/local/lib/python3.6/site-packages/milvus/client/grpc_handler.py:468: BaseException
------------------------------ Captured log setup ------------------------------
[2020-08-25 16:15:22-INFO-root]: {'index_type': 'BIN_FLAT', 'params': {'nlist': 128}} (test_load_collection.py:40)
----------------------------- Captured stderr call -----------------------------
Error: <BaseException: (code=5, message=Index metric type SUBSTRUCTURE does not match index type BIN_FLAT)>
------------------------------ Captured log call -------------------------------
[2020-08-25 16:15:22-INFO-root]: JACCARD (test_load_collection.py:68)
[2020-08-25 16:15:23-INFO-root]: HAMMING (test_load_collection.py:68)
[2020-08-25 16:15:25-INFO-root]: TANIMOTO (test_load_collection.py:68)
[2020-08-25 16:15:26-INFO-root]: SUBSTRUCTURE (test_load_collection.py:68)
```

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Method of installation**
- [ ] Docker/cpu
- [x] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)


**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/core/src/server/ValidationUtil.cpp
+++ b/core/src/server/ValidationUtil.cpp
@@ -354,25 +354,10 @@ ValidateSegmentRowCount(int64_t segment_row_count) {
 
 Status
 ValidateIndexMetricType(const std::string& metric_type, const std::string& index_type) {
-    static std::set<std::string> s_valid_metric = {
-        milvus::knowhere::Metric::L2,
-        milvus::knowhere::Metric::IP,
-        milvus::knowhere::Metric::HAMMING,
-        milvus::knowhere::Metric::JACCARD,
-        milvus::knowhere::Metric::TANIMOTO,
-        milvus::knowhere::Metric::SUBSTRUCTURE,
-        milvus::knowhere::Metric::SUPERSTRUCTURE,
-    };
-
-    if (index_type == knowhere::IndexEnum::INDEX_FAISS_IDMAP) {
-        if (s_valid_metric.find(metric_type) == s_valid_metric.end()) {
-            std::string msg =
-                "Invalid index metric type: " + metric_type + ". " + "Make sure the metric type is in MetricType list.";
-            LOG_SERVER_ERROR_ << msg;
-            return Status(SERVER_INVALID_INDEX_METRIC_TYPE, msg);
-        }
-    } else if (index_type == knowhere::IndexEnum::INDEX_FAISS_BIN_IVFFLAT ||
-               index_type == knowhere::IndexEnum::INDEX_FAISS_BIN_IDMAP) {
+    if (index_type == knowhere::IndexEnum::INDEX_FAISS_IDMAP ||
+        index_type == knowhere::IndexEnum::INDEX_FAISS_BIN_IDMAP) {
+        // pass
+    } else if (index_type == knowhere::IndexEnum::INDEX_FAISS_BIN_IVFFLAT) {
         // binary
         if (metric_type != knowhere::Metric::HAMMING && metric_type != knowhere::Metric::JACCARD &&
             metric_type != knowhere::Metric::TANIMOTO) {

--- a/tests/milvus_python_test/test_index.py
+++ b/tests/milvus_python_test/test_index.py
@@ -585,9 +585,11 @@ class TestIndexBinary:
         # insert 6000 vectors
         ids = connect.insert(binary_collection, binary_entities)
         connect.flush([binary_collection])
-
-        with pytest.raises(Exception) as e:
+        if get_l2_index["index_type"] == "BIN_FLAT":
             res = connect.create_index(binary_collection, binary_field_name, get_l2_index)
+        else:
+            with pytest.raises(Exception) as e:
+                res = connect.create_index(binary_collection, binary_field_name, get_l2_index)
 
     """
     ******************************************************************

